plugin/plugin.go

```diff
@@ -404,7 +451,25 @@ func NotifyFlush(dom *domain.Domain, pluginName string) error {
 	if p == nil || p.Manifest.flushWatcher == nil || p.State != Ready {
 		return errors.Errorf("plugin %s doesn't exists or unsupported flush or doesn't start with PD", pluginName)
 	}
-	_, err := dom.GetEtcdClient().KV.Put(context.Background(), p.Manifest.flushWatcher.path, "")
+	_, err := dom.GetEtcdClient().KV.Put(context.Background(), p.Manifest.flushWatcher.path, strconv.Itoa(int(p.Disabled)))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ChangeDisableFlagAndFlush changes plugin disable flag and notify other nodes to do same change..
+func ChangeDisableFlagAndFlush(dom *domain.Domain, pluginName string, disable bool) error {
+	p := getByName(pluginName)
+	disableInt := uint32(0)
+	if disable {
+		disableInt = 1
+	}
+	atomic.StoreUint32(&p.Disabled, disableInt)
+	if p == nil || p.Manifest.flushWatcher == nil || p.State != Ready {
```

Comment:
Should move `if p == nil` after line#463 to avoid nil point panic in line#468 ?

Reply:
done~~